README.md

```diff
@@ -166,6 +166,205 @@ This solution collects anonymous operational metrics to help AWS improve the
 quality of features of the solution. For more information, including how to disable
 this capability, please see the [implementation guide](https://docs.aws.amazon.com/solutions/latest/amazon-marketing-cloud-uploader-from-aws/collection-of-operational-metrics.html).
 
+---
+# Advanced Usage
+
+## Integration with solution for file upload automation
+
+This solution's `start_amc_transformation` API endpoint can be integrated with other ETL pipelines to provide with the built-in normalization and timeseries partitioning before uploading data to Amazon Marketing Cloud (AMC).
+
+The purpose of the endpoint is to normalize and hash clear-text PII and partition time series data sets for AMC. Timestamp columns must be formatted before using the endpoint according to ISO 8601.
+
+The `start_amc_transformation` API endpoint starts an AWS Glue Job with the arguments passed into it. The following is a summary of the arguments included in a request to the endpoint and expected outputs of the API:
+
+### Inputs
+
+- `countryCode`: string
+	- The country code to be used for different country-specific normalizations to apply to all rows in the dataset
+	- 2-digit ISO country code
+	- Ex: `"countryCode": "US"`
+	
+- `datasetId`: string
+	- The name of the new data set that will be created with the data being uploaded
+	- It can also be the name of an existing data set that the data will be uploaded to, though the schema should match (`deletedFields`, `piiFields`, and column names should be the same)
+	- Ex: `"datasetId": "new-AMC-dataset"`
+	
+- `deletedFields`: string (array)
+	- The list of columns to be deleted or dropped from the original data set prior to uploading to AMC
+	- Ex: `"deletedFields": "[\"product_quantity\",\"product_name\"]"`
+	
+	- Can be an empty array string if no data needs to be deleted
+	- Ex: `"deletedFields": "[]"`
+	
+- `destination_endpoints`: string (array)
+	- The list of AMC endpoints to receive of uploads
+	- Ex: `destination_endpoints: "[\" https://rrhcd93zj3.execute-api.us-east-1.amazonaws.com/prod\"]"`
+	
+- `outputBucket`: string
+	- The name of the ETL artifact S3 bucket where the normalized output data will be written to
+	- Ex: `"outputBucket": "myTestStack-etl-artifacts"`,
+	
+- `period`: string
+	- The time period of the data set
+	- If uploding a dimension type data set, this will be autodetected and set to `P1D`
```

Comment:
this argument actually is not used if uploading a dimension data set (it only applies to time series data). We should mention that it is required for fact data and should be omitted for dimension data